File: makeup_city/makeup_city/report/employee_discount_limit_inquiry/employee_discount_limit_inquiry.js
// Copyright (c) 2025, Createch Solutions and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Employee Discount Limit Inquiry"] = {
	"filters": [
		{
			"fieldname": "from_date",
			"label": "From Date",
			"fieldtype": "Date",
			"default": frappe.datetime.month_start(),
			"read_only": 1
		},
		{
			"fieldname": "to_date",
			"label": "To Date",
			"fieldtype": "Date",
			"default": frappe.datetime.month_end(),
			"read_only": 1
		},
		{
			"fieldname": "customer",
			"label": "Customer",
			"fieldtype": "Link",
			"options": "Customer",
			"default": null,
			"reqd": 1,
			"get_query": () => {
				return {
					filters: {
						customer_group: "Employee"
					}
				};
			}
		}
	]
};

File: makeup_city/makeup_city/report/employee_discount_limit_inquiry/employee_discount_limit_inquiry.py
import frappe
from frappe.utils import flt

def execute(filters=None):
    columns = [
        {"label": "Employee Name", "fieldname": "customer", "fieldtype": "Link", "options": "Customer", "width": 200},
        {"label": "Employee ID", "fieldname": "customer_name", "fieldtype": "Data", "width": 250},
        {"label": "Monthly Limit", "fieldname": "monthly_limit", "fieldtype": "Currency", "width": 200},
        {"label": "Availed Discount", "fieldname": "availed_discount", "fieldtype": "Currency", "width": 200},
        {"label": "Remaining Limit", "fieldname": "remaining_limit", "fieldtype": "Currency", "width": 200},
    ]

    conditions = """
        si.docstatus = 1 AND si.customer_group = 'Employee'
    """

    if filters and filters.get("customer"):
        conditions += " AND si.customer = %(customer)s"

    data = frappe.db.sql(f"""
        SELECT
            si.customer,
            si.customer_name,
            c.custom_monthly_employee_discount_limit AS monthly_limit,
            SUM(
                IFNULL(si.discount_amount, 0) + (
                    SELECT SUM(IFNULL(sii.discount_amount, 0))
                    FROM `tabSales Invoice Item` sii
                    WHERE sii.parent = si.name
                )
            ) AS availed_discount
        FROM `tabSales Invoice` si
        LEFT JOIN `tabCustomer` c ON c.name = si.customer
        WHERE {conditions}
        GROUP BY si.customer, si.customer_name, c.custom_monthly_employee_discount_limit
    """, filters, as_dict=True)

    for row in data:
        row["remaining_limit"] = flt(row.monthly_limit) - flt(row.availed_discount)

    return columns, data
